worker-queue-rabbit/src/main/java/com/hpe/caf/worker/queue/rabbit/WorkerConfirmListener.java

```diff
@@ -77,32 +78,49 @@ public void handleAck(long sequenceNo, boolean multiple)
         throws IOException
     {
         LOG.debug("RabbitMQ broker ACKed published sequence id {} (multiple: {})", sequenceNo, multiple);
-        handle(sequenceNo, multiple, ConsumerAckEvent::new);
+        handle(sequenceNo, multiple, t -> {
+            t.incrementAcknowledgementCount();
+            if(t.areAllResponsesAcknowledged()){
+                return new ConsumerAckEvent(Long.valueOf(t.getInboundMessageId()));
+            }
+            return null;
+        });
     }
 
     @Override
     public void handleNack(long sequenceNo, boolean multiple)
         throws IOException
     {
         LOG.warn("RabbitMQ broker NACKed published sequence id {} (multiple: {})", sequenceNo, multiple);
-        handle(sequenceNo, multiple, ConsumerRejectEvent::new);
+        handle(sequenceNo, multiple, t -> {
+               if (!t.isNegativeAckSent()) {
+                   t.markNegativeAckAsSent();
+                   return new ConsumerRejectEvent(Long.valueOf(t.getInboundMessageId()));
+               }
+               return null;
+           });
     }
 
-    private void handle(long sequenceNo, boolean multiple, Function<Long, Event<QueueConsumer>> eventSource)
+    private void handle(long sequenceNo, boolean multiple, Function<RabbitTaskInformation, Event<QueueConsumer>> eventSource)
     {
+        //TODO Andy Correct this logic
```

Comment:
Is this TODO ready to be removed? Are there others?

Reply:
Nothing else is left, can be removed.

Reply:
Please remove it